jabgui/src/main/java/org/jabref/gui/libraryproperties/general/GeneralPropertiesViewModel.java

```diff
@@ -199,21 +199,31 @@ private Path getBrowseDirectory(String configuredDir) {
         if (configuredDir.isEmpty()) {
             return preferences.getFilePreferences().getWorkingDirectory();
         }
-        Optional<Path> foundPath = this.databaseContext.getFileDirectories(preferences.getFilePreferences()).stream()
-                                                       .filter(path -> path.toString().endsWith(configuredDir))
-                                                       .filter(Files::exists).findFirst();
 
-        if (foundPath.isEmpty()) {
+        Optional<Path> libPath = this.databaseContext.getDatabasePath();
+
+        if (libPath.isEmpty()) {
             dialogService.notify(Localization.lang("Path %0 could not be resolved. Using working dir.", configuredDir));
             return preferences.getFilePreferences().getWorkingDirectory();
         }
-        return foundPath.get();
+
+        Path path = libPath.get().getParent().resolve(configuredDir).normalize();
+
+        if (!Files.isDirectory(path)) {
+            dialogService.notify(Localization.lang("Path %0 could not be resolved. Using working dir.", configuredDir));
+            return preferences.getFilePreferences().getWorkingDirectory();
+        }
+        return path;
     }
 
     private ValidationMessage validateDirectory(String directoryPath, String messageKey) {
         try {
-            Path path = Path.of(directoryPath);
-            if (!Files.isDirectory(path)) {
+            Optional<Path> libPath = this.databaseContext.getDatabasePath();
+
+            if (libPath.isEmpty() || !Files.isDirectory(libPath.get()
+                                   .getParent()
+                                   .resolve(directoryPath)
+                                   .normalize())) {
```

Comment:
Use the power of optionals wisely:

```suggestion
if (!libPath.map(p -> p.getParent().resolve(directoryPath).normalize())
           .map(Files::isDirectory)
           .orElse(false)) {
```